lib/web_ui/lib/src/engine/skwasm/skwasm_impl/paint.dart

```diff
@@ -8,15 +8,26 @@ import 'package:ui/src/engine.dart';
 import 'package:ui/src/engine/skwasm/skwasm_impl.dart';
 import 'package:ui/ui.dart' as ui;
 
-class SkwasmPaint implements ui.Paint {
-  factory SkwasmPaint() {
-    return SkwasmPaint._fromHandle(paintCreate());
+class SkwasmPaint implements SkwasmObjectWrapper<RawPaint>, ui.Paint {
+  factory SkwasmPaint() => SkwasmPaint._fromHandle(paintCreate());
+
+  SkwasmPaint._fromHandle(this.handle) {
+    _registry.register(this);
+  }
+
+  void dispose() {
+    assert(!_isDisposed);
+    _registry.unregister(this);
```

Comment:
This pattern is used in all types. I wonder it can be rolled into `SkwasmFinalizationRegistry`. We could have a nicely symmetric API:

```dart
registry.register(this);
registry.dispose(this); // also unregisters since it doesn't need to be GC'd any more
```